python/maths/Bailey_borwein_plouffe.py

```diff
@@ -0,0 +1,46 @@
+def bailey_borwein_plouffe(digit_position: int, precision: int = 1000) -> str:
+
+    if (not isinstance(digit_position, int)) or (digit_position <= 0):
+        raise ValueError("Digit position must be a positive integer")
+    elif (not isinstance(precision, int)) or (precision < 0):
+        raise ValueError("Precision must be a nonnegative integer")
+
+    sum_result = (
+        4 * _subsum(digit_position, 1, precision)
+        - 2 * _subsum(digit_position, 4, precision)
+        - _subsum(digit_position, 5, precision)
+        - _subsum(digit_position, 6, precision)
+    )
+
+    return hex(int((sum_result % 1) * 16))[2:]
+
+
+def _subsum(
+    digit_pos_to_extract: int, denominator_addend: int, precision: int
+) -> float:
+    Private helper function to implement the summation
```

Comment:
This line is giving error. Please compile the program before committing. Do the necessary changes. Thankyou :)